packages/dds/task-manager/src/taskManager.ts

```diff
@@ -478,10 +501,21 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         this.connectionWatcher.on("disconnect", disconnectHandler);
         this.completedWatcher.on("completed", checkIfCompleted);
 
-        if (!this.connected) {
+        if (!this.isAttached()) {
+            // Simulate auto-ack in detached scenario
+            assert(this.clientId !== undefined, "clientId is undefined");
+            this.addClientToQueue(taskId, this.clientId);
+            // If we volunteered using placeHolderClientId, then we need to watch for when we connect and are assigned a
+            // clientId post-attach. At that point we should re-enter the queue with a real volunteer op.
+            if (this.clientId === placeholderClientId) {
+                this.connectionWatcher.once("connect", () => {
```

Comment:
We might get a disconnect/connect while still remaining detached, in which case we should avoid sending the op